torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -84,7 +130,22 @@ struct TORCH_API AccumulateGrad : public Node {
         if (new_grad.is_sparse()) {
           update_grad(new_grad.clone());
         } else {
-          update_grad(new_grad.clone(at::MemoryFormat::Contiguous));
+          // Deep copies new_grad according to the "Gradient Layout Contract."
+          if (variable.is_non_overlapping_and_dense()) {
+            // (1)
+            if (variable.strides() == new_grad.strides()) {
+              update_grad(new_grad.clone(at::MemoryFormat::Preserve));
```

Comment:
For strided Tensors, this won't have the expected effect right? As the result from the clone will be rowmajor contiguous and may not have the same stride as the variable.

Reply:
I thought the whole point of `tensor.clone(at::MemoryFormat::preserve)` was to produce a clone with strides matching `tensor`.  Is that wrong?

Reply:
It will preserve strides for non-overlapping dense tensors.

Reply:
that's exactly what i need :)

Reply:
@albanD does this resolve your concerns or am I misunderstanding?

Reply:
Yes, I should have read the doc about `preserve` before!